custom_components/versatile_thermostat/__init__.py

```diff
@@ -236,7 +237,9 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry):
 
         thermostat_type = config_entry.data.get(CONF_THERMOSTAT_TYPE)
 
-        if thermostat_type == CONF_THERMOSTAT_CENTRAL_CONFIG:
+        old_version = config_entry.version * 100 + config_entry.minor_version
+
+        if old_version < 201 and thermostat_type == CONF_THERMOSTAT_CENTRAL_CONFIG:
```

Comment:
BTW @jmcollin78 are you ok with this change? It is not related to the unique_id itself, but it looks strange to me that we always reset those values in case of migration.